### PR TITLE
TCA default all pools for a borrower

### PR DESCRIPTION
**Pull request by yuchenlintt**

### CODE REVIEW

File: contracts/truefi2/BorrowingMutex.sol [hunk 1]
```diff
@@ -43,11 +43,11 @@ contract BorrowingMutex is IBorrowingMutex, UpgradeableClaimable {
     function unlock(address borrower) external override {
         address _locker = locker[borrower];
         require(_locker == msg.sender, "BorrowingMutex: Only locker can unlock");
-        locker[borrower] = address(0);
+        locker[borrower] = address(uint256(Status.Unlocked));
```
Comment (vanruch) [inline contracts/truefi2/BorrowingMutex.sol:46]: I don't think IBorrowingMutex is a good place for this. We can keep default address const in this file and Blocked address in TCA as it's not strictly connected to the mutex logic.

File: contracts/truefi2/TrueCreditAgency.sol [hunk 2]
```diff
@@ -424,57 +424,60 @@ contract TrueCreditAgency is UpgradeableClaimable, ITrueCreditAgency {
...
+            _rebucket(pool, borrower, oldScore, newScore, 0);
 
-        loanFactory.createDebtToken(pool, borrower, principal.add(_interest));
+            uint256 _interest = interest(pool, borrower);
+            borrowerTotalPaidInterest[pool][borrower] = borrowerTotalPaidInterest[pool][borrower].add(_interest);
+            poolTotalPaidInterest[pool] = poolTotalPaidInterest[pool].add(_interest);
 
-        if (totalBorrowed(borrower) == 0) {
-            borrowingMutex.unlock(borrower);
+            loanFactory.createDebtToken(pool, borrower, principal.add(_interest));
```
Comment (vanruch) [inline contracts/truefi2/TrueCreditAgency.sol:475]: Should we really deploy the tokens if the borrower didn't take anything from the pool?

File: contracts/truefi2/BorrowingMutex.sol (hunk 1 above)
Comment (yuchenlintt) [reply contracts/truefi2/BorrowingMutex.sol]: Good point, it's better not to expose implementation details in the interface like this.

I moved both Unlocked and Banned into the BorrowingMutex impl as constants and added a `ban()` function to the interface.

File: contracts/truefi2/TrueCreditAgency.sol (hunk 2 above)
Comment (yuchenlintt) [reply contracts/truefi2/TrueCreditAgency.sol]: Thanks for catching this! I remember my previous version had a debt == 0 check, but I must have forgotten to add it back in.

File: test/truefi2/lines-of-credit/TrueCreditAgency.test.ts [hunk 3]
```diff
@@ -1064,80 +1046,43 @@ describe('TrueCreditAgency', () => {
...
+      it('only creates DebtTokens for pools with nonzero debt', async () => {
+        await creditAgency.connect(borrower).repayInFull(usdcPool.address)
+        const debtTokens = await extractDebtTokens(creditAgency.enterDefault(borrower.address))
+        expect(debtTokens.length).to.eq(1)
+        expect(await debtTokens[0].pool()).to.eq(tusdPool.address)
       })
-    })
-  })
 
-  describe('pokeBorrowLimitTimer', () => {
```
Comment (yuchenlintt) [inline test/truefi2/lines-of-credit/TrueCreditAgency.test.ts:1105]: If you expand the code below, you'll see this section has been duplicated (probably due to merge conflict resolutions).

File: test/truefi2/lines-of-credit/TrueCreditAgency.test.ts [hunk 4]
```diff
@@ -966,33 +966,15 @@ describe('TrueCreditAgency', () => {
     })
 
     describe('reverts if borrower', () => {
-      it('pool is not supported', async () => {
```
Comment (yuchenlintt) [inline test/truefi2/lines-of-credit/TrueCreditAgency.test.ts:969]: I removed 3 duplicate tests (merge conflicts?) that are all unnecesary now that we're iterating through supported pools ourselves.

**Review by vanruch — APPROVED** [review]

File: contracts/truefi2/BorrowingMutex.sol [hunk 5]
```diff
@@ -24,6 +27,11 @@ contract BorrowingMutex is IBorrowingMutex, UpgradeableClaimable {
 
     event BorrowerUnlocked(address borrower, address locker);
 
+    modifier onlyCanLock() {
```
Comment (skalermo) [inline contracts/truefi2/BorrowingMutex.sol:30]: I feel like `onlyAllowedToLock` would be a better name for this modifier.

**Review by MiksuJak — APPROVED** [review]

**Review by skalermo — APPROVED** [review]

**Review by MiksuJak — APPROVED** [review]

**Pull request merged**